Several PHP 7.4 fixes

brainfoolong opened the pull request:
On a fresh 3.13.0 install with php 7.4.0 following error occurs when tracker does track a page view, when using Mysqli as database driver. 

`Notice: Trying to access array offset on value of type int in /srv/0x.at/matomo/core/Tracker/Db/Mysqli.php on line 50`

This fix this error

brainfoolong committed:
fixed php 7.4 warning

On a fresh 3.13.0 install with php 7.4.0 following error occurs when tracker does track a page view, when using Mysqli as database driver. 

`Notice: Trying to access array offset on value of type int in /srv/0x.at/matomo/core/Tracker/Db/Mysqli.php on line 50`

This fix this error

brainfoolong committed:
Fix Php 7.4 warning array_kex_exists with object

This error appears, this change fixes this issue

`DataTable/Filter/ColumnDelete.php(164): Deprecated - array_key_exists(): Using array_key_exists() on objects is deprecated. Use isset() or property_exists() instead - Matomo 3.11.0`

tsteur commented on core/DataTable/Filter/ColumnDelete.php:
```diff
@@ -161,7 +161,7 @@ protected function removeColumnsFromTable(&$table)
             foreach ($this->columnsToRemove as $column) {
 
-                if (!array_key_exists($column, $row)) {
+                if (!property_exists($column, $row)) {
```
Could you maybe undo this change? We will investigate this as part of https://github.com/matomo-org/matomo/issues/15210

brainfoolong replied:
Yes, done.

brainfoolong committed:
reverted fix

As suggested here
https://github.com/matomo-org/matomo/pull/15222/files/cba4f22e12a23ec392f45decceedc49e5753eea5#diff-a5f249b6134f80011f165be56c5995b1

This wasn't a proper fix, i got other errors after applying that.

tsteur commented:
Awesome, thanks for quick change. Very appreciated.

tsteur reviewed (approved):
Need to wait for tests before merging though
